torch/csrc/jit/custom_class.h

```diff
@@ -0,0 +1,193 @@
+
+#pragma once
+
+#include <ATen/core/function_schema.h>
+#include <ATen/core/ivalue.h>
+#include <ATen/core/jit_type.h>
+#include <ATen/core/op_registration/op_registration.h>
+#include <ATen/core/stack.h>
+#include <c10/util/C++17.h>
+#include <c10/util/Metaprogramming.h>
+#include <c10/util/TypeList.h>
+#include <pybind11/pybind11.h>
+#include <torch/csrc/jit/operator.h>
+#include <torch/csrc/jit/pybind_utils.h>
+#include <torch/csrc/jit/script/compilation_unit.h>
+#include <torch/csrc/jit/tracer.h>
+#include <torch/csrc/utils/variadic.h>
+#include <iostream>
+#include <sstream>
+
+namespace py = pybind11;
+namespace torch {
+namespace jit {
+
+namespace detail {
+template <class R, class...>
+struct types {
+  constexpr static bool hasRet = true;
+  using type = types;
+};
+template <class... args>
+struct types<void, args...> {
+  constexpr static bool hasRet = false;
+  using type = types;
+};
+template <class Sig>
+struct args;
+template <class R, class CurClass, class... Args>
+struct args<R (CurClass::*)(Args...)> : types<R, Args...> {};
+template <class Sig>
+using args_t = typename args<Sig>::type;
+} // namespace detail
+template <class... Types>
+detail::types<void, Types...> init() {}
+
+template <class CurClass>
+class class_ {
+  std::string className;
+  std::shared_ptr<py::class_<CurClass>> pyClass = nullptr;
```

Comment:
Hmm, I'm perhaps not familiar with what the standards are. What should this be instead?

I made it a `shared_ptr` because it doesn't have a default constructor and initializing it in the initialization list would have made for a pretty unwieldy expression.

Reply:
Ok, a unique_ptr or a c10::optional might be more appropriate then.